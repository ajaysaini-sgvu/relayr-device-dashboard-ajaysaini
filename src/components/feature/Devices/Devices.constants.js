const DEVICES_CONSTANTS = {
  GET_DEVICES_LIST: "GET_DEVICES_LIST",
  SET_DEVICES_LIST_SUCCESS: "SET_DEVICES_LIST_SUCCESS",
  SET_DEVICES_LIST_FAILURE: "SET_DEVICES_LIST_FAILURE",
  UPDATE_DEVICE_STATUS_REQUEST: "UPDATE_DEVICE_STATUS_REQUEST",
  SET_UPDATE_DEVICE_STATUS_SUCCESS: "SET_UPDATE_DEVICE_STATUS_SUCCESS",
  SET_UPDATE_DEVICE_STATUS_FAILURE: "SET_UPDATE_DEVICE_STATUS_FAILURE"
};

export default DEVICES_CONSTANTS;
