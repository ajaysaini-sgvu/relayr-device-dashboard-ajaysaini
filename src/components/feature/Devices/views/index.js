export { default } from "./Devices.view";
