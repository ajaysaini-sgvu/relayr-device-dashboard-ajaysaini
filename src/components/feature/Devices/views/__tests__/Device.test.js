import React from "react";
import { shallow, configure } from "enzyme";
import Devices from "../Devices.view";

const Adapter = require("enzyme-adapter-react-16");

configure({ adapter: new Adapter() });

describe("Devices", () => {
  it("should render correctly", () => {
    const props = {
      devices: [
        {
          name: "acceleration_x",
          unit: "m/s2",
          value: 25.993848858558,
          timestamp: new Date().getTime() - Math.floor(Math.random() * 100000),
          active: true
        }
      ]
    };
    const component = shallow(<Devices {...props} />);
    expect(component).toMatchSnapshot();
  });

  it("should render error", () => {
    const props = {
      devices: [],
      showUpdatedNotification: "Something went wrong"
    };
    const component = shallow(<Devices {...props} />);
    expect(component).toMatchSnapshot();
  });

  it("test updateDeviceStatus", () => {
    const props = {
      onDeviceClick: jest.fn()
    };
    const component = shallow(<Devices {...props} />);
    const mockedEvent = { target: {}, preventDefault: jest.fn() };
    component.instance().updateDeviceStatus(mockedEvent, { name: "test" });
    expect(props.onDeviceClick).toHaveBeenCalledTimes(1);
  });

  it("test handleSearchInput", () => {
    const component = shallow(<Devices />);
    const mockedEvent = {
      target: { value: "test" },
      preventDefault: jest.fn()
    };
    component.find("input").simulate("change", mockedEvent);
    expect(component.state("input")).toBe("test");
  });
});
