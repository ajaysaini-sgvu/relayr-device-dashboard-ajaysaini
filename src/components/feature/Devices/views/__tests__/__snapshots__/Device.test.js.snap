// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Devices should render correctly 1`] = `ShallowWrapper {}`;

exports[`Devices should render error 1`] = `ShallowWrapper {}`;
